SSA/Projects/LLVMRiscV/select.lean

```diff
@@ -0,0 +1,108 @@
+import SSA.Projects.LLVMRiscV.PeepholeRefine
+import SSA.Projects.LLVMRiscV.simpproc
+import SSA.Projects.RISCV64.Tactic.SimpRiscV
+
+open LLVMRiscV
+set_option Elab.async true
+
+/-
+This file contains the instruction lowering for the `llvm.select` instruction.
+
+LLVM version 11.0 lowers `select` to a conditional branch regardless of the
+optimization level used. However, our current setup does not support branches.
+
+To handle this, we manually encode the semantics of the `select` instruction
+using bitwise operations. While this results in a longer RISC-V instruction
+sequence compared to LLVM's native lowering, it allows us to support lowering
+the instruction within our framework.
+-/
+
+/- # select, RiscV (leveraging bit-wise operations)  -/
+def select_riscv := [LV| {
+  ^entry (%cond : i1, %arg0: i64, %arg1: i64):
+    %0 = "builtin.unrealized_conversion_cast" (%arg0) : (i64) -> (!i64)
+    %1 = "builtin.unrealized_conversion_cast" (%arg1) : (i64) -> (!i64)
+    %2 = "builtin.unrealized_conversion_cast" (%cond) : (i1) -> (!i64)
+    %3 = slli %2, 63 : !i64
+    %4 = srai %3, 63 : !i64 --propagating the condition bit to all bits within the 64 bit vector
+    %5 = and %0, %4 : !i64 -- mask the value_true with the condition
+    %6 = li (18446744073709551615) : !i64
+    %7 = xor %6, %4: !i64
+    %8 = and %7, %1 : !i64 -- mask the value_false with the condition
+    %9 = or %5, %8 : !i64 -- return either value_true or value_false
+    %10 = "builtin.unrealized_conversion_cast" (%9) : (!i64) -> (i64)
+    llvm.return %10 : i64
+  }]
+
+def select_llvm_64 := [LV| {
+    ^entry (%cond : i1, %arg0: i64, %arg1: i64):
+    %0 = "llvm.select"(%cond, %arg0, %arg1) <{"fastmathFlags" = #llvm.fastmath<none>}> : (i1, i64, i64) -> i64
+    llvm.return %0 : i64
+  }]
+
+def select_riscv_select_llvm_64 : LLVMPeepholeRewriteRefine 64 [Ty.llvm (.bitvec 64), Ty.llvm (.bitvec 64), Ty.llvm (.bitvec 1)] where
+  lhs:= select_llvm_64
+  rhs:= select_riscv
+  correct := by
+    unfold select_riscv select_llvm_64
+    simp_peephole
+    simp_riscv
+    simp [LLVM.select]
+    simp_alive_case_bash
+    case value.poison.poison =>
+      intro x
+      split <;> simp
+    case value.poison.value =>
+      intro x x'
+      split <;> simp <;> bv_decide
+    case value.value.poison =>
+      intro x x'
+      split <;> simp <;> bv_decide
+    case value.value.value =>
+      intro x x' x''
+      split <;> simp <;> bv_decide
```

Comment:
you have two approaches: 
1. separate each case and have `simp only ...` and `bv decide` there (each case will start with `\.`, e.g. [here](https://github.com/opencompl/lean4/blob/1fd7206f0038e851b645422ec1dc77f75b104387/src/Init/Data/BitVec/Lemmas.lean#L115))
2. if you `simp?` at this point you'll see all the theorems used in all branches: you can put all them together in the same `simp only` - the theorems that are not necessary in some branc will be ignored. However, this might not work, as some rewrites might overlap in weird ways in `simp`. Give it a shot :)

Reply:
this lgtm now, but I'd like to wait and see if @alexkeizer has any sugggestions to make the proof more coincise :)